backend/src/common/types/service.type.ts

```diff
@@ -1,10 +1,15 @@
+import { type UserUpdateProfileRequestDto } from 'shared';
+
 type Service<T = unknown> = {
     find(query: Record<string, T>): Promise<T>;
     findAll(): Promise<{
         items: T[];
     }>;
     create(payload: unknown): Promise<T>;
-    update(): Promise<T>;
+    update(
+        id: number,
+        updatedDetails: UserUpdateProfileRequestDto,
```

Comment:
I think we shouldn't set UserUpdateProfileRequestDto in global service type because we implement Service in many places. For example, for goals or workouts we need to update some data and we will get an error because in update we should pass UserUpdateProfileRequestDto.